src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,108 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] arrayList;
+    private int currentCapacity;
+    private int currentIndex;
+
+    public ArrayList() {
+        arrayList = (T[]) new Object[DEFAULT_CAPACITY];
+        currentCapacity = DEFAULT_CAPACITY;
+    }
 
     @Override
     public void add(T value) {
-
+        addElementToArray(value, currentIndex);
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > currentIndex) {
+            throw new ArrayIndexOutOfBoundsException(" Index above the size of Array");
+        }
+        sizeCheck(index);
+        if (index < currentIndex) {
+            System.arraycopy(arrayList, index, arrayList, index + 1, currentIndex + 1);
```

Comment:
put into separate method

Reply:
I just thought - on this line I'm just calling a method. Is there a value to have a method that just calls another one? But you are probably right (in case we decided to replace it with our own solution) .